lib/identity/client.js

```diff
@@ -1348,6 +1350,106 @@ class Client extends PartialClient {
+          'Content-Type': 'application/json',
+          'X-Tozny-Session': sessionToken,
+        },
+        body: JSON.stringify({ identity_id: toznyId }),
+      }
+    )
+    await checkStatus(response)
+    const data = await response.json()
+    return InitiateWebAuthnChallengeData.decode(data)
+  }
+
+  /**
+   * `registerWebAuthnDevice` finalizes the registration of a WebAuthn MFA device for this identity.
+   * @param {PublicKeyCredential} publicKeyCredential this is the response from a call to `navigator.credentials.create`
+   * @param {string} deviceName a name for this device.
+   * @param {string} tabId the id from the initiation challenge call. used to track session across api calls.
+   * @returns {IdentityMFADevices}
+   */
+  // Note: Supporting cross identity registration? Add an optional identity id arg w/ fallback to this client.
+  async registerWebAuthnDevice(publicKeyCredential, deviceName, tabId) {
+    const sessionToken = await this.agentToken()
+    const toznyId = this.storage.config.clientId
+    const response = await this.storage.authenticator.tsv1Fetch(
+      `${this.storage.config.apiUrl}/v1/identity/mfa`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Tozny-Session': sessionToken,
+        },
+        body: JSON.stringify({
+          identity_id: toznyId,
+          tab_id: tabId,
+          mfa_devices: {
+            webauthn:
+              InitiateWebAuthnChallengeData.convertPublicKeyCredentialToRegistrationData(
+                publicKeyCredential,
+                deviceName
+              ),
+          },
+        }),
+      }
+    )
+    await checkStatus(response)
+    const data = await response.json()
+    return IdentityMFADevices.decode(data)
+  }
+
+  /**
+   * `searchIdentityMFADeviceCredentials` finds details of MFA devices for the identities matching the `searchParams`.
+   * If no `searchParams` are provided, it will query for the current identity's MFA devices.
+   * Only realm admins have permission to query the details of other identities' MFA devices.
+   * @param {string} realmName Name of realm
+   * @param {object} [searchParams] Data containing info on which users to find MFA devices for. defaults to this client's identity.
+   *                                All search parameters are taken as to be an OR query.
+   * @param {string[]} [searchParams.toznyIds] List of Tozny client ids to search over
+   * @param {string[]} [searchParams.userIds] List of user ids to search over
+   * @returns {IdentityMFADevices}
+   */
+  async searchIdentityMFADeviceCredentials(realmName, searchParams) {
+    const toznyIds = (searchParams && searchParams.toznyIds) || []
+    const userIds = (searchParams && searchParams.userIds) || []
+    const searchIsEmpty =
+      !searchParams || (toznyIds.length === 0 && userIds.length === 0)
+    if (searchIsEmpty) {
+      // fallback to this client's identity if search is empty.
```

Comment:
i love this

Reply:
works so long as you dont accidentally pass it to the server as `tonzyIds` 😂